polytope_server/common/observability/otel.py

```diff
@@ -0,0 +1,86 @@
+from opentelemetry import trace
+from opentelemetry.propagate import inject, extract
+from contextlib import contextmanager
+import logging
+
+def add_trace_context(request):
+    carrier = {}
+
+    # Inject the context into the carrier
+    inject(carrier)
+
+    # Adding trace_id and span_id to the request data for logging
+    request.otel_trace_ctx = {
+        'carrier': carrier  # Store the injected carrier
+    }
+
+    current_span = trace.get_current_span()
+    current_span.set_attribute("polytope.request.id", request.id)
+
+    logging.debug(f"[OTEL] Request created with carrier: {request.otel_trace_ctx['carrier']}")
+
+def update_trace_context(request):
+    carrier = {}
+
+    # Inject the context into the carrier
+    inject(carrier)
+
+    # Adding trace_id and span_id to the request data for logging
+    request.otel_trace_ctx = {
+        'carrier': carrier  # Store the injected carrier
+    }
+
+    logging.debug(f"[OTEL] Request created with carrier: {request.otel_trace_ctx['carrier']}")
+
+
+def restore_trace_context(request):
```

Comment:
Rename to extract_trace_context?

Reply:
the function’s primary goal is to re-establish the context for subsequent operations (and not just extracting data), I believe better suits restore